### DISCUSSION TITLE
bin/rails tailwindcss:watch via foreman exiting with 0 status

### DISCUSSION THREAD

**Discussion by pedz** [tailwindlabs/tailwindcss#9831] [Help] [answered]
This is a rather convoluted set up I suppose.  I have a new Rails 7 project with --css=tailwindcss inside a Docker container.  This version of Rails implements a `bin/dev` script which installs foreman (the Ruby gem) and executes foreman with a Project.dev file.  The contents are:
```
web: bin/rails server -b 0.0.0.0 -p 3000
css: bin/rails tailwindcss:watch
```
I have `bin/dev` as CMD in the Dockerfile.  The container starts up, starts the server and tailwindcss.  Tailwindcss exists with 0 status which triggers foreman to kill the other processes and things stop.  Curiously, I do not get back to my command prompt but the Docker container has exited (and the rails server has exited).  A ^C gets me back to the laptop's command prompt.

The command being executed is:
```
["/usr/local/bundle/gems/tailwindcss-rails-2.0.21-aarch64-linux/exe/aarch64-linux/tailwindcss", "-i", "/rails-7-passkey-demo/app/assets/stylesheets/application.tailwind.css", "-o", "/rails-7-passkey-demo/app/assets/builds/tailwind.css", "-c", "/rails-7-passkey-demo/config/tailwind.config.js", "--minify", "-w"]
```

If I execute the same `bin/rails tailwindcss:watch` from a command prompt within the container, it executes (I'm guessing -- I'm new to tailwindcss) the way it should.  There is a few update lines such as how much time it took and I do **not** get back to my prompt in the container.  I can touch one of the files and tailwindcss loops again as I assume it should.

I've added various debugging such as a wrapper script around the executable to print out the arguments and environment.  The arguments in both cases are the same.  The environment differs in trivial ways.  e.g. PORT and PS are set via foreman.

My current work around is to comment out the css line in the foreman procfile and then run the tailwindcss watch by hand in another window.

tailwindcss v3.2.4
rails 7.0.4
tailwindcss-rails (2.0.21-aarch64-linux)
cat /etc/os-release
```
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by pedz** [comment]
`bin/rails tailwindcss:watch` as a separate container also exists with a 0 status rather than waiting for file system events.

**Comment by metkat** [comment, marked as answer]
Adding `tty: true` to the compose file that the tailwindcss:watch process is running in will prevent the immediate exit. 

The [docs for tailwindcss-rails also claim you can pass `always` to the process](https://github.com/rails/tailwindcss-rails#update-assets-automatically) which I think would be a better solution, but this appears to only work if there's also a tty assigned to the container so maybe kind of pointless?
`

**Reply by pedz** [reply]
I'm finally getting back to this project.

Both solutions work: Adding the `tty: true` line or using `command: [ "rails", "tailwindcss:watch[always]" ]`
